akka-docs-new/src/main/paradox/guide/modules.md

```diff
@@ -0,0 +1,165 @@
+ 
+* How do I restore the state of an entity/actor when system restarts or crashes?
+* How do I implement a [CQRS system](https://msdn.microsoft.com/en-us/library/jj591573.aspx)?
+* How do I ensure reliable delivery of messages in face of network errors and system crashes?
+* How do I introspect domain events that has lead an entity to its current state?
+
+### Cluster Singleton
+
+A common (in fact, a bit too common) use case in distributed systems is to have a single entity responsible 
+for a given task which is shared among other members of the cluster and migrated if the host system fails. 
+While this undeniably introduces a common bottleneck for the whole cluster that limits scaling,
+there are scenarios where the use of this pattern is unavoidable. Cluster singleton allows a cluster to elect an 
+actor system which will host a particular actor while other systems can always access said service independently from 
+where it is.
+
+The Singleton module can be used to solve these problems:
+
+* How do I ensure that only one instance is running of a service in the whole cluster?
+* How do I ensure that the service is up even if the system hosting it currently crashes or shut down during the process of scaling down?
+* How do I reach this instance from any member of the cluster assuming that it can migrate to other systems over time?
+
+### Cluster Sharding
+
+Persistence solves the problem of restoring an actor’s state from persistent storage after system restart or crash. 
+It does not solve itself the problem of distributing a set of such actors among members of an Akka cluster. 
+Sharding is a pattern that mostly used together with Persistence to balance a large set of persistent entities 
+(backed by actors) to members of a cluster and also migrate them to other systems if one of the members crash.
+ 
+The problem space that Sharding targets:
+
+* How do I model and scale out a large set of stateful entities on a set of systems?
+* How do I ensure that entities in the cluster are distributed properly so that load is properly balanced across the machines?
+* How do I ensure migrating entities from a crashed system without losing state?
+* How do I ensure that an entity does not exist on multiple systems at the same time and hence kept consistent?
+
+### Cluster Publish-Subscribe
+
+For coordination among systems it is often necessary to distribute messages to all, or one system of a set of 
+interested systems in a cluster. This pattern is usually called publish-subscribe and this module solves this exact 
+problem. It is possible to subscribe to topics and receive messages published to that topic and it is also possible 
+to broadcast or anycast messages to subscribers of that topic.
+
+* How do I broadcast messages to an interested set of parties in a cluster?
+* How do I anycast messages to a member from an interested set of parties in a cluster?
+* How to subscribe and unsubscribe for events of a certain topic in the cluster?
+
```

Comment:
Can you write the summary? I don't have any good ideas to introduce DData without confusion (compared to persistence/sharding)